allennlp/data/dataset_readers/hugging_face_datasets_reader.py

```diff
@@ -0,0 +1,148 @@
+        self.dataset_name = dataset_name
+        self.config_name = config_name
+        self.index = -1
+
+        if config_name:
+            self.dataset = load_dataset(self.dataset_name, self.config_name, split=split)
+        else:
+            self.dataset = load_dataset(self.dataset_name, split=split)
+
+    def _read(self, file_path) -> Iterable[Instance]:
+        """
+        Reads the instances from the given `file_path` and returns them as an
+        `Iterable`.
+
+        You are strongly encouraged to use a generator so that users can
+        read a dataset in a lazy way, if they so choose.
+        """
+        for entry in self.dataset:
+            yield self.text_to_instance(entry)
+
+    def text_to_instance(self, *inputs) -> Instance:
+        """
+        Does whatever tokenization or processing is necessary to go from textual input to an
+        `Instance`.  The primary intended use for this is with a
+        :class:`~allennlp.predictors.predictor.Predictor`, which gets text input as a JSON
+        object and needs to process it to be input to a model.
+
+        The intent here is to share code between :func:`_read` and what happens at
+        model serving time, or any other time you want to make a prediction from new data.  We need
+        to process the data in the same way it was done at training time.  Allowing the
+        `DatasetReader` to process new text lets us accomplish this, as we can just call
+        `DatasetReader.text_to_instance` when serving predictions.
+
+        The input type here is rather vaguely specified, unfortunately.  The `Predictor` will
+        have to make some assumptions about the kind of `DatasetReader` that it's using, in order
+        to pass it the right information.
+        """
+
+        # Features indicate the different information available in each entry
+        features = self.dataset.features
+        fields = dict()
+
+        # TODO we need to support all different datasets features of https://huggingface.co/docs/datasets/features.html
+        for feature in features:
+            value = features[feature]
+
+            # datasets ClassLabel maps to LabelField
+            if isinstance(value, ClassLabel):
+                field = LabelField(inputs[0][feature], skip_indexing=True)
+
+            # datasets Value can be of different types
+            elif isinstance(value, Value):
+
+                # String value maps to TextField
+                if value.dtype == 'string':
+                    # Since TextField has to be made of Tokens add whole text as a token
+                    field = TextField([Token(inputs[0][feature])])
+
+                # Since we don't have a numerical equivalent use default LabelField
+                elif value.dtype == 'int32':
+                    field = LabelField(inputs[0][feature], skip_indexing=True)
+                else:
+                    field = LabelField(inputs[0][feature], skip_indexing=True)
```

Comment:
field is assigned same value at lines 91 and 93 irrespective of else blocks they belong to